src/main/java/frc/robot/subsystems/Drivetrain.java

```diff
@@ -454,89 +380,204 @@ public void runChassisPID(double x, double y, double rot) {
+    return m_yController;
+  }
+  public PIDController getRotationController() {
+    return m_rotationController;
+  }
+
+  // PIDs for Pathplanner
+  public PIDController getPathplannerXController() {
+    return m_pathplannerXController;
+  }
+  public PIDController getPathplannerYController() {
+    return m_pathplannerYController;
+  }
+  public PIDController getPathplannerRotationController() {
+    return m_pathplannerRotationController;
+  }
+
+  public PIDController getBalanceController() {
+    return m_balancePID;
+  }
+
   /**
-   * Sets up feedforward savers.
+   * Gets the current robot pose from the pose estimator.
    */
-  private void setUpFeedforwardSavers() {
-    m_driveStaticFeedForwardSaver = new Double[] {
-      m_modules[0].getDriveFeedForwardKS(),
-      m_modules[1].getDriveFeedForwardKS(),
-      m_modules[2].getDriveFeedForwardKS(),
-      m_modules[3].getDriveFeedForwardKS()
-    };
-    m_driveVelFeedForwardSaver = new Double[] {
-      m_modules[0].getDriveFeedForwardKV(),
-      m_modules[1].getDriveFeedForwardKV(),
-      m_modules[2].getDriveFeedForwardKV(),
-      m_modules[3].getDriveFeedForwardKV()
-    };
-    m_steerStaticFeedForwardSaver = new Double[] {
-      m_modules[0].getSteerFeedForwardKS(),
-      m_modules[1].getSteerFeedForwardKS(),
-      m_modules[2].getSteerFeedForwardKS(),
-      m_modules[3].getSteerFeedForwardKS()
-    };
-    m_steerVelFeedForwardSaver = new Double[] {
-      m_modules[0].getSteerFeedForwardKV(),
-      m_modules[1].getSteerFeedForwardKV(),
-      m_modules[2].getSteerFeedForwardKV(),
-      m_modules[3].getSteerFeedForwardKV()
-    };
+  public Pose2d getPose() {
+    return m_poseEstimator.getEstimatedPosition();
   }
-  
-  public Double[] getDriveStaticFeedforwardArray() {
-    return m_driveStaticFeedForwardSaver;
+
+  /**
+   * @return the yaw of the robot, aka heading, the direction it is facing
+   */
+  public Rotation2d getYaw() {
+    return m_poseEstimator.getEstimatedPosition().getRotation();
   }
-  public Double[] getDriveVelocityFeedforwardArray() {
-    return m_driveVelFeedForwardSaver;
+
+  /**
+   * @return the pitch of the robot
+   */
+  public Rotation2d getPitch() {
+    return Rotation2d.fromDegrees(m_pigeon.getPitch());
   }
-  public Double[] getSteerStaticFeedforwardArray() {
-    return m_steerStaticFeedForwardSaver;
+
+  /**
+   * @return the roll of the robot
+   */
+  public Rotation2d getRoll() {
+    return Rotation2d.fromDegrees(m_pigeon.getRoll());
   }
-  public Double[] getSteerVelocityFeedforwardArray() {
-    return m_steerVelFeedForwardSaver;
+
+  /**
+   * @return chassis speed of swerve drive
+   */
+  public ChassisSpeeds getChassisSpeeds() {
+    return DriveConstants.kKinematics.toChassisSpeeds(getModuleStates());
+  }
+
+  public ChassisSpeeds getFieldRelativeChassisSpeeds() {
+    return ChassisSpeeds.fromFieldRelativeSpeeds(
+      getChassisSpeeds(),
+      getPose().getRotation());
```

Comment:
getYaw() could be more clear

Reply:
added where the yaw is from in java doc.

Reply:
I mean instead of doing getPose().getRotation() you can just do getYaw() which is probably more clear anyway